provider-service/vai/cmd/main.go

```diff
@@ -26,66 +27,72 @@ func main() {
 	if err != nil {
 		panic(err)
 	}
+
 	ctx := logr.NewContext(context.Background(), logger)
 
-	config, err := configLoader.LoadConfig(ctx)
+	serviceConfig, err := baseConfig.LoadConfig(baseConfig.Config{
+		Server: baseConfig.Server{
+			Host: "0.0.0.0",
+			Port: 8080,
+		},
+	})
 	if err != nil {
 		panic(err)
 	}
+	logger.Info(fmt.Sprintf("loaded base config: %+v", serviceConfig))
 
-	k8sClient, err := NewK8sClient()
+	vaiProviderConfig, err := baseConfig.LoadConfig(vaiConfig.VAIProviderConfig{Name: serviceConfig.ProviderName})
 	if err != nil {
+		logger.Error(err, "failed to load provider config", "provider", serviceConfig.ProviderName, "namespace", serviceConfig.Pod.Namespace)
 		panic(err)
 	}
+	logger.Info(fmt.Sprintf("loaded provider config: %+v", vaiProviderConfig), "provider", serviceConfig.ProviderName, "namespace", serviceConfig.Pod.Namespace)
 
-	vaiConfig := vaiConfig.VAIProviderConfig{
-		Name: config.ProviderName,
-	}
+	go runServer(ctx, vaiProviderConfig, serviceConfig)
```

Comment:
I moved the code that does `server.Start()` in a function (because it's identical to the kfp code and that's what we do there), but `servcer.Start()` wasn't running as a goroutine before, even though it is in kfp.

So I made it run in a goroutine, but is there a reason not to?

Reply:
prevents any server blow up causing the eventing to also be blocked from running